src/connection.ts

```diff
@@ -3538,34 +3538,41 @@ class Connection extends EventEmitter {
         let credentials: TokenCredential;
 
-        switch (authentication.type) {
-          case 'token-credential':
-            credentials = authentication.options.credential;
-            break;
-          case 'azure-active-directory-password':
-            credentials = new UsernamePasswordCredential(
-              authentication.options.tenantId ?? 'common',
-              authentication.options.clientId,
-              authentication.options.userName,
-              authentication.options.password
-            );
-            break;
-          case 'azure-active-directory-msi-vm':
-          case 'azure-active-directory-msi-app-service':
-            const msiArgs = authentication.options.clientId ? [authentication.options.clientId, {}] : [{}];
-            credentials = new ManagedIdentityCredential(...msiArgs);
-            break;
-          case 'azure-active-directory-default':
-            const args = authentication.options.clientId ? { managedIdentityClientId: authentication.options.clientId } : {};
-            credentials = new DefaultAzureCredential(args);
-            break;
-          case 'azure-active-directory-service-principal-secret':
-            credentials = new ClientSecretCredential(
-              authentication.options.tenantId,
-              authentication.options.clientId,
-              authentication.options.clientSecret
-            );
-            break;
+        try {
+          switch (authentication.type) {
+            case 'token-credential':
+              credentials = authentication.options.credential;
+              break;
+            case 'azure-active-directory-password':
+              credentials = new UsernamePasswordCredential(
+                  authentication.options.tenantId ?? 'common',
+                  authentication.options.clientId,
+                  authentication.options.userName,
+                  authentication.options.password
+              );
+              break;
+            case 'azure-active-directory-msi-vm':
+            case 'azure-active-directory-msi-app-service':
+              const msiArgs = authentication.options.clientId ? [authentication.options.clientId, {}] : [{}];
+              credentials = new ManagedIdentityCredential(...msiArgs);
+              break;
+            case 'azure-active-directory-default':
+              const args = authentication.options.clientId ? { managedIdentityClientId: authentication.options.clientId } : {};
+              credentials = new DefaultAzureCredential(args);
+              break;
+            case 'azure-active-directory-service-principal-secret':
+              credentials = new ClientSecretCredential(
+                  authentication.options.tenantId,
+                  authentication.options.clientId,
+                  authentication.options.clientSecret
+              );
+              break;
+          }
+        } catch (err) {
+          signal.throwIfAborted();
```

Comment:
Is this required? I don't see any async processing happening between here and the previous signal check? 🤔

Reply:
I have no idea - I was just copying the same logic for the other catch incase it was important.